### ISSUE TITLE
精度复现存在些许差异

### ISSUE THREAD

**Issue by keithAND2020** [PuFanqi23/MonoDGP#14]
作者您好，这是一篇非常优秀的论文，目前我在复现您的代码，在采用和您相同的config下，我跑出来的结果和您的结果有些许差异，您在logs/里面，

<img width="1137" height="847" alt="Image" src="https://github.com/user-attachments/assets/d88f5064-e964-4ae4-9f16-2354bbae2e78" />
是这样的结果。我在相同的配置下，结果如下：

<img width="1129" height="819" alt="Image" src="https://github.com/user-attachments/assets/ce6cef1a-0ce9-4dc3-b980-766d67fc2e84" />

想请教一下是什么原因呢？谢谢！
下面是配置
random_seed: 444

dataset:
  type: &dataset_type 'KITTI'
  root_dir: './data/KITTIDataset/'
  train_split: 'train'
  test_split: 'val'
  batch_size: 8
  use_3d_center: True
  class_merging: False
  use_dontcare: False
  bbox2d_type: 'anno'  # 'proj' or 'anno'
  meanshape: False  # use predefined anchor or not
  writelist: ['Car']
  clip_2d: False

  aug_pd: True
  aug_crop: True

  random_flip: 0.5
  random_crop: 0.5
  random_mixup3d: 0.5
  scale: 0.05
  shift: 0.05

  depth_scale: 'normal'

model_name: 'monodgp'
model:
  num_classes: 3
  return_intermediate_dec: True
  device: 'cuda'

 Backbone
  backbone: 'resnet50'
  train_backbone: True
  num_feature_levels: 4
  dilation: False
  position_embedding: 'sine'  # 'sine' or 'learned'
  masks: False

Depth predictor
  mode: LID
  num_depth_bins: 80
  depth_min: 1e-3
  depth_max: 60.0

Transformer
  with_box_refine: True
  use_dn: False
  init_box: False
  enc_layers: 3
  dec_layers: 3
  hidden_dim: 256
  dim_feedforward: 256
  dropout: 0.1
  nheads: 8
  num_queries: 50
  group_num: 11
  enc_n_points: 4
  dec_n_points: 4
DN
  scalar: 5
  label_noise_scale: 0.2
  box_noise_scale: 0.4
  num_patterns: 0
Loss
  aux_loss: True

Loss coefficients
  cls_loss_coef: 2
  focal_alpha: 0.25

  bbox_loss_coef: 5
  giou_loss_coef: 2
  3dcenter_loss_coef: 10
  dim_loss_coef: 1
  angle_loss_coef: 1
  depth_loss_coef: 1
  depth_map_loss_coef: 1
  region_loss_coef: 1

Matcher
  set_cost_class: 2
  set_cost_bbox: 5
  set_cost_giou: 2
  set_cost_3dcenter: 10

optimizer:
  type: 'adamw'
  lr: 0.0002
  weight_decay: 0.0001

lr_scheduler:
  type: 'step'  # 'step' or 'cos'
  warmup: False  # 5 epoches, cosine warmup, init_lir=0.00001 in default
  decay_rate: 0.5
  decay_list: [85, 125, 165, 205]

trainer:
  max_epoch: 250
  gpu_ids: '1'
  save_frequency: 1  # checkpoint save interval (in epoch)
  #resume_model: True
  #pretrain_model: path/to/ckpt
  save_path: 'outputs/'
  save_all: False  # False for only savng the best and latest ckpts
  use_dn: False
  scalar: 5
  label_noise_scale: 0.2
  box_noise_scale: 0.4
  num_patterns: 0

tester:
  type: *dataset_type
  mode: single  # 'single' or 'all'; 'all' to evaluate all ckpts in the folder
  checkpoint: 195  # the start ckpt epoch for 'all'
  threshold: 0.2  # confidence filter
  topk: 50

**Comment by PuFanqi23** [comment]
你好，精度复现存在小范围偏差是正常现象，仓库里上传的checkpoint是在多次训练中选取的表现较好的三次，可以多训练几次看看效果怎么样。

**Comment by jiashengsu418** [comment]
> 作者您好，这是一篇非常优秀的论文，目前我在复现您的代码，在采用和您相同的config下，我跑出来的结果和您的结果有些许差异，您在logs/里面，
> 
> <img alt="Image" width="1137" height="847" src="https://private-user-images.githubusercontent.com/57097690/464000190-d88f5064-e964-4ae4-9f16-2354bbae2e78.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg2NDc1NDEsIm5iZiI6MTc1ODY0NzI0MSwicGF0aCI6Ii81NzA5NzY5MC80NjQwMDAxOTAtZDg4ZjUwNjQtZTk2NC00YWU0LTlmMTYtMjM1NGJiYWUyZTc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIzVDE3MDcyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlYzdkODdhYzU4NDM0YThiODczYTQxNGFjYWZmYzQzYmQwZDJlZmFjNTRmOTIzMzEyZjY2NmQ3YzAzMGI3NTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q8RoanYpuqeHmoHc9XEgYxipf2KLMPo4k9KBtXqPvyM"> 是这样的结果。我在相同的配置下，结果如下： <img alt="Image" width="1129" height="819" src="https://private-user-images.githubusercontent.com/57097690/464000354-ce6cef1a-0ce9-4dc3-b980-766d67fc2e84.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg2NDc1NDEsIm5iZiI6MTc1ODY0NzI0MSwicGF0aCI6Ii81NzA5NzY5MC80NjQwMDAzNTQtY2U2Y2VmMWEtMGNlOS00ZGMzLWI5ODAtNzY2ZDY3ZmMyZTg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIzVDE3MDcyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2ZWE2ZWU1NjkyMGVhODdhN2Y2MTEwNWIzOWZlOWE4YmQxMjM0Njc3YTZjZDhhZjBjNjIwMTBkNmU0YjMzYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4Vm6vfo29hJfhXfHnm8Wy-N7eBKhlC0-M_YYyC_c2yk">
> 想请教一下是什么原因呢？谢谢！ 下面是配置 random_seed: 444
> 
> dataset: type: &dataset_type 'KITTI' root_dir: './data/KITTIDataset/' train_split: 'train' test_split: 'val' batch_size: 8 use_3d_center: True class_merging: False use_dontcare: False bbox2d_type: 'anno' # 'proj' or 'anno' meanshape: False # use predefined anchor or not writelist: ['Car'] clip_2d: False
> 
> aug_pd: True aug_crop: True
> 
> random_flip: 0.5 random_crop: 0.5 random_mixup3d: 0.5 scale: 0.05 shift: 0.05
> 
> depth_scale: 'normal'
> 
> model_name: 'monodgp' model: num_classes: 3 return_intermediate_dec: True device: 'cuda'
> 
> Backbone backbone: 'resnet50' train_backbone: True num_feature_levels: 4 dilation: False position_embedding: 'sine' # 'sine' or 'learned' masks: False
> 
> Depth predictor mode: LID num_depth_bins: 80 depth_min: 1e-3 depth_max: 60.0
> 
> Transformer with_box_refine: True use_dn: False init_box: False enc_layers: 3 dec_layers: 3 hidden_dim: 256 dim_feedforward: 256 dropout: 0.1 nheads: 8 num_queries: 50 group_num: 11 enc_n_points: 4 dec_n_points: 4 DN scalar: 5 label_noise_scale: 0.2 box_noise_scale: 0.4 num_patterns: 0 Loss aux_loss: True
> 
> Loss coefficients cls_loss_coef: 2 focal_alpha: 0.25
> 
> bbox_loss_coef: 5 giou_loss_coef: 2 3dcenter_loss_coef: 10 dim_loss_coef: 1 angle_loss_coef: 1 depth_loss_coef: 1 depth_map_loss_coef: 1 region_loss_coef: 1
> 
> Matcher set_cost_class: 2 set_cost_bbox: 5 set_cost_giou: 2 set_cost_3dcenter: 10
> 
> optimizer: type: 'adamw' lr: 0.0002 weight_decay: 0.0001
> 
> lr_scheduler: type: 'step' # 'step' or 'cos' warmup: False # 5 epoches, cosine warmup, init_lir=0.00001 in default decay_rate: 0.5 decay_list: [85, 125, 165, 205]
> 
> trainer: max_epoch: 250 gpu_ids: '1' save_frequency: 1 # checkpoint save interval (in epoch) #resume_model: True #pretrain_model: path/to/ckpt save_path: 'outputs/' save_all: False # False for only savng the best and latest ckpts use_dn: False scalar: 5 label_noise_scale: 0.2 box_noise_scale: 0.4 num_patterns: 0
> 
> tester: type: *dataset_type mode: single # 'single' or 'all'; 'all' to evaluate all ckpts in the folder checkpoint: 195 # the start ckpt epoch for 'all' threshold: 0.2 # confidence filter topk: 50

您好，我想问一下可以复现出来了吗

**Comment by lemony1988** [comment]
请问您这个是test结果吗？需要更改monodgp.ymal中的test_spilt和train_spilt吗？
